pandas/core/reshape/pivot.py

```diff
@@ -422,13 +423,18 @@ def _convert_by(by):
 
 @Substitution("\ndata : DataFrame")
 @Appender(_shared_docs["pivot"], indents=1)
-def pivot(data: "DataFrame", index=None, columns=None, values=None) -> "DataFrame":
+def pivot(
+    data: "DataFrame",
+    index: Optional[Union[Label, List[Optional[Label]]]] = None,
```

Comment:
```suggestion
    index: Optional[Union[Label, Sequence[Optional[Label]]]] = None,
```

For API items we want to be as generic as possible, so `Sequence` instead of `List` and `Mapping` instead of `Dict` (unless the interface really isn't generic)

Reply:
Same comment on next two lines

Reply:
> For API items we want to be as generic as possible, so `Sequence` instead of `List` and `Mapping` instead of `Dict` (unless the interface really isn't generic)

thanks for the very nice tip!!! 👍